pkg/operator/config/config.go

```diff
@@ -122,20 +130,25 @@ func (c *Config) InjectTrustedCA() bool {
 	return len(strings.TrimSpace(c.TrustedCAConfigMapName)) != 0
 }
 
+// RequeuePeriod returns requeue period as time.Duration.
+func (c *Config) RequeuePeriod() time.Duration {
+	return time.Duration(c.RequeuePeriodSeconds) * time.Second
+}
+
 // isOCP returns true if the platform is OCP
-func isOCP(kubeClient discovery.DiscoveryInterface) bool {
+func isOCP(kubeClient discovery.DiscoveryInterface) (bool, error) {
 	// Since, CRD for OpenShift API Server was introduced in OCP v4.x we can verify if the current cluster is on OCP v4.x by
 	// ensuring that resource exists against Group(operator.openshift.io), Version(v1) and Kind(OpenShiftAPIServer)
 	// In case it doesn't exist we assume that external dns is running on non OCP 4.x environment
 	resources, err := kubeClient.ServerResourcesForGroupVersion(openshiftResourceGroup + "/" + openshiftResourceVersion)
 	if err != nil {
-		return false
+		return false, err
```

Comment:
@thejasn : I saw the wrong OpenShift detection on my local cluster too (remember the case in the CI). This should help to propagate the error if there was one.

Reply:
Looks like it didn't fail on the CI (maybe rerun, not sure).

Reply:
Yeah, now if there is a problem during the platform detection the container will exit with non-zero status which will make it a subject to the restart. Since OLM makes sure the deployment of the operator has the minimum availability - the installation won't succeed until the platform is detected. The error I got on my local cluster was:
```
2022-05-30T14:44:17.293Z	ERROR	setup	failed to detect the platform	{"error": "Get \"https://10.217.4.1:443/apis/operator.openshift.io/v1?timeout=32s\": dial tcp 10.217.4.1:443: i/o timeout"}

$ oc get svc -o wide -A | grep 10.217.4.1
default                                            kubernetes                                 ClusterIP      10.217.4.1     <none> 
```
So, I suppose that this could be the problem for the CI clusters too, some temporary API unavailability which used to be interpreted as `Not OpenShift` case.